src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/ReadStringDataArray.cpp

```diff
@@ -0,0 +1,135 @@
+
+using namespace nx::core;
+
+namespace
+{
+constexpr int32_t k_RBR_FILE_NOT_OPEN = -1000;
+constexpr int32_t k_RBR_READ_EOF = -1030;
+constexpr int32_t k_RBR_READ_ERROR = 1040;
+constexpr int32_t k_RBR_FILE_NOT_EXIST = 1050;
+
+constexpr size_t k_BufferSize = 10000;
+
+std::string ConvertDelimiterIndexToString(uint64 index)
+{
+  switch(index)
+  {
+  case nx::core::read_string_data_array::k_CommaIndex:
+    return {","};
+  case nx::core::read_string_data_array::k_SemicolonIndex:
+    return {";"};
+  case nx::core::read_string_data_array::k_SpaceIndex:
+    return {" "};
+  case nx::core::read_string_data_array::k_ColonIndex:
+    return {":"};
+  case nx::core::read_string_data_array::k_TabIndex:
+    return {"\t"};
+  case nx::core::read_string_data_array::k_NewLineIndex:
+    return {"\n"};
+  }
+  return {""};
+}
+} // End anonymous namespace
+
+namespace nx::core
+{
+ReadStringDataArray::ReadStringDataArray(DataStructure& dataStructure, const IFilter::MessageHandler& mesgHandler, const std::atomic_bool& shouldCancel,
+                                         const ReadStringDataArrayInputValues* inputValues)
+: m_DataStructure(dataStructure)
+, m_MessageHandler(mesgHandler)
+, m_ShouldCancel(shouldCancel)
+, m_InputValues(inputValues)
+{
+}
+
+ReadStringDataArray::~ReadStringDataArray() noexcept = default;
+
+Result<> ReadStringDataArray::operator()()
+{
+  auto data = m_DataStructure.getDataRefAs<StringArray>(m_InputValues->outputArrayPath);
+  char delimiter = nx::core::CsvParser::IndexToDelimiter(m_InputValues->delimiterIndex);
+
+  if(!fs::exists(m_InputValues->inputFileValue))
+  {
+    return MakeErrorResult(k_RBR_FILE_NOT_EXIST, fmt::format("Input file does not exist: {}", m_InputValues->inputFileValue.string()));
+  }
+
+  std::ifstream in(m_InputValues->inputFileValue.c_str(), std::ios_base::in | std::ios_base::binary);
+  if(!in.is_open())
+  {
+    return MakeErrorResult(k_RBR_FILE_NOT_OPEN, fmt::format("Could not open file for reading: {}", m_InputValues->inputFileValue.string()));
+  }
+
+  in.imbue(std::locale(std::locale(), new nx::core::CsvParser::DelimiterType(delimiter)));
+
+  //  std::array<char, k_BufferSize> buf = {};
+  //  char* buffer = buf.data();
+  std::string headerLine;
+  // Skip some header line by just reading those bytes into the pointer knowing that the next
+  // thing we are going to do it over write those bytes with the real data that we are after.
+  for(int i = 0; i < m_InputValues->skipLineCount; i++)
+  {
+    std::getline(in, headerLine);
+    if(in.fail())
+    {
+      return MakeErrorResult(k_RBR_READ_ERROR, fmt::format("Could not read data from file while skipping header lines: {}", m_InputValues->inputFileValue.string()));
+    }
+  }
+
+  usize numTuples = data.getNumberOfTuples();
+  usize scalarNumComp = data.getNumberOfComponents();
+
+  usize totalSize = numTuples * scalarNumComp;
+  usize lineNum = m_InputValues->skipLineCount;
+  using StringVector = std::vector<std::string>;
```

Comment:
```suggestion
```
Since this is not a dependent type and only used in one place, we probably don't need it.